File: shared/domain/models/invitation.go
package models

import "time"

// InvitationType は招待の種別。
type InvitationType string

const (
	InvitationTypeGroupJoin   InvitationType = "group_join"   // LinkSelfグループ招待
	InvitationTypeRolePromote InvitationType = "role_promote" // ロール任命
)

// InvitationStatus は招待の状態。
type InvitationStatus string

const (
	InvitationStatusPending  InvitationStatus = "pending"
	InvitationStatusAccepted InvitationStatus = "accepted"
	InvitationStatusDeclined InvitationStatus = "declined"
)

// Invitation は招待・任命を表す。
type Invitation struct {
	ID          string           `json:"id"`
	Type        InvitationType   `json:"type"`
	Status      InvitationStatus `json:"status"`
	InviterID   string           `json:"inviterId"`  // 招待者のDID
	InviteeID   string           `json:"inviteeId"`  // 対象者のDID
	TargetRole  Role             `json:"targetRole"` // 任命先ロール
	Description string           `json:"description"`
	CreatedAt   time.Time        `json:"createdAt"`
	ResolvedAt  *time.Time       `json:"resolvedAt"`
}

File: pwa/src/domain/models/invitation.ts
// 招待・任命のドメインモデル。
// 仕様: docs/wants/04_メンバー管理と権限.md
// 参照実装: shared/domain/models/invitation.go

import type { Role } from "./user";

/** 招待の種別。 */
export type InvitationType =
  | "group_join" // LinkSelfグループ招待
  | "role_promote"; // ロール任命

/** 招待の状態。 */
export type InvitationStatus = "pending" | "accepted" | "declined";

/** 招待・任命。 */
export interface Invitation {
  id: string;
  type: InvitationType;
  status: InvitationStatus;
  /** 招待者のDID */
  inviterId: string;
  /** 対象者のDID */
  inviteeId: string;
  /** 任命先ロール */
  targetRole: Role;
  description: string;
  /** ISO 8601 */
  createdAt: string;
  /** ISO 8601 */
  resolvedAt: string | null;
}
